lib/transform/pathRegexTransformer.ts

```diff
@@ -77,7 +85,7 @@ const buildPathRegex = (
       hasMultiPathParam = true;
     } else {
       hostTemplate = hostTemplate.replace("{" + v + "}", ":" + i);
-      path = path.replace("{" + v + "}", ":" + i);
+      path = path.replace("{" + v + "}", `:${i}(\\d+)`);
```

Comment:
Is there any unit test for this that could add a regression check for this

Reply:
A full regression run is literally a re-run of pretty much the whole `azure-rest-api-specs` repo, so at the very least none of the existing public specs don't have this issue.

I'll add a test for it but I wanted to get the PR out and regression builds running.